x/application/keeper/msg_server_delegate_to_gateway.go

```diff
@@ -36,6 +36,13 @@ func (k msgServer) DelegateToGateway(goCtx context.Context, msg *types.MsgDelega
 		return nil, sdkerrors.Wrapf(types.ErrAppGatewayNotFound, "gateway not found with address: %s", msg.GatewayAddress)
 	}
 
+	// Ensure the application is not already delegated to the maximum number of gateways
+	maxDelegatedParam := k.GetParams(ctx).MaxDelegatedGateways
+	if int64(len(app.DelegateeGatewayPubKeys)) >= maxDelegatedParam {
+		logger.Info("Application already delegated to maximum number of gateways: %d", maxDelegatedParam)
```

Comment:
Am I understanding correctly that this log would print on the poktrolld node? If so, are we sure we want that? As a sequencer or supplier (i.e. poktrolld node operator) I wouldn't expect to see logs from message handler functions which are effectively user error.

Reply:
**tl;dr I think it's okay to leave as is but open to w/e approach you think is best.**

THEORETICALLY this SHOULD be an event that anyone can capture.

However, in practice, looking at validator/sequencer nodes is how things actually get debugged (from personal experience), so I think it's reasonable.

For example, the tx error would be what we return below and the log line is some telimitary the sequencer operator can use to understand that someone is either missing or doesn't know how to interact with the protocol.